### DISCUSSION TITLE
Are Milvus ids signed or unsigned 64bit integers?

### DISCUSSION THREAD

**Discussion by iantbutler01** [milvus-io/milvus#4623] [Q&A and General discussion] [answered]
I know they need to be positive, but are they signed or unsigned?

**Comment by shiyu22** [comment, marked as answer]
Hi Ian, as you say, ids must be non-negative 64-bit integers, but they are signed.

And what scenarios are you using milvus for? Is there a requirement for the ids?

**Reply by iantbutler01** [reply]
Hi there,

Thank you for the answer. We are using Milvus as our primary vector database for semantic similarity for a programming focused search engine. 

As for your second question yes, right now because we cannot store text information alongside the vectors in Milvus we have to store that additional metadata somewhere else and then retrieve it by the ids that Milvus returns with the query results. We manually set our ids because to keep that parity between our metadata store and Milvus. Right now we're using a bit shifted sha256 hash of the url, dropping the last bit with the hopes that collision would still be fairly rare but it would be nicer to be able to either store the text data alongside the vectors or be able to use something like a UUID for the ids.
